### ISSUE TITLE
[microprofile] @CircuitBreaker with configuration file does not override the requestVolumeThreshold.

### ISSUE THREAD

**Issue by charleech** [payara/Payara#3762]
# Description #
----------

When define the `@CircuitBreaker: requestVolumeThreshold` via the `META-INF/microprofile-config.properties` with other value than the default as `20`, it still be `20`.

## Expected Outcome

The `requestVolumeThreshold` should be overridden by the value defined by `META-INF/microprofile-config.properties` respectively.

## Current Outcome

The `requestVolumeThreshold`  has been a default value as `20`.

## Steps to reproduce (Only for bug reports) 

1. Annotate the method with empty `@CircuitBreaker`
2. Set the CircuitBreaker attribute via the `META-INF/microprofile-config.properties` as `requestVolumeThreshold=4`
3. Access the service via cURL with failure 5 times, there is no any `CircuitBreakerOpenException` thrown.

### Samples
```properties
# META-INF/microprofile-config.properties
my.service.MyService/alwaysFail/CircuitBreaker/failOn=java.lang.Throwable.class
my.service.MyService/alwaysFail/CircuitBreaker/requestVolumeThreshold=4
my.service.MyService/alwaysFail/CircuitBreaker/failureRatio=0.75d
my.service.MyService/alwaysFail/CircuitBreaker/successThreshold=2
my.service.MyService/alwaysFail/CircuitBreaker/delay=1000
```
```java
package my.service;
@ApplicationPath("/myapi")
public class MyApplication extends Application {
}
```
```java
package my.service;
@Path("ft")
@ApplicationScoped
public class MyResource {
    @GET
    @CircuitBreaker
    public Response alwaysFail() {
        throws new IllegalStateException("This always fails.");
    }
}
```

```bash
curl -v -X GET \
http://localhost:8080/myweb/myapi/ft

# Even I try to execute the above curl for 5 times the response will be
# "java.lang.IllegalStateException: This always fails."
# The CircuitBreakerOpenException has been thrown at 21st round.
```

Furthermore, I also have a chance to look at the [FaultToleranceService](https://github.com/payara/Payara/blob/bc93c5e8948d3cb4a94c321a407d2589cf9cf074/appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceService.java#L451) and found that the method [registerCircuitBreaker](https://github.com/payara/Payara/blob/bc93c5e8948d3cb4a94c321a407d2589cf9cf074/appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceService.java#L470) use the `requestVolumeThreshold` value from the annotation directly instead of using the read from the configuration file as

``` java
faultToleranceObjects.get(applicationName).getCircuitBreakerStates().get(invocationTarget)
                    .put(fullMethodSignature, new CircuitBreakerState(circuitBreaker.requestVolumeThreshold()));
``` 
This may be the root cause.

## Context (Optional)
I've tried to learn the Microprofile: Fault Tolerance and tried to config them via the `META-INF/microprofile-config.properties`.

## Environment ##

From Docker `payara/micro:latest`

- **Payara Version**: `Payara Micro  5.184 #badassmicrofish (build 89)`
- **Edition**: `Micro`
- **JDK Version**: 
```
openjdk version "1.8.0_181"
OpenJDK Runtime Environment (IcedTea 3.9.0) (Alpine 8.181.13-r0)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)
```
- **Operating System**: 
```
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.8.1
PRETTY_NAME="Alpine Linux v3.8"
```
- **Database**:  N/A

**Comment by charleech** [comment]
To update the program flow.

1. The [CircuitBreakerInterceptor.circuitBreak()](https://github.com/payara/Payara/blob/05cab4c6f075075ed870d91d87ff0fce1d005889/appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/interceptors/CircuitBreakerInterceptor.java#L232) try to get or register as the following

```java
        int requestVolumeThreshold = (Integer) FaultToleranceCdiUtils.getOverrideValue(
                config, CircuitBreaker.class, "requestVolumeThreshold", invocationContext, Integer.class)
                .orElse(circuitBreaker.requestVolumeThreshold());

        ....

        CircuitBreakerState circuitBreakerState = faultToleranceService.getCircuitBreakerState(appName,
                invocationContext.getTarget(),
                invocationContext.getMethod(), circuitBreaker);
```

2. The [FaultToleranceService.getCircuitBreakerState()](https://github.com/payara/Payara/blob/bc93c5e8948d3cb4a94c321a407d2589cf9cf074/appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceService.java#L415) determines if it is existed or not. If not, register it as the following: -

```java
        if (annotatedMethodCircuitBreakerStates == null) {
            circuitBreakerState = registerCircuitBreaker(applicationName, invocationTarget, fullMethodSignature, 
                    circuitBreaker);
        } else {
            circuitBreakerState = annotatedMethodCircuitBreakerStates.get(fullMethodSignature);

            if (circuitBreakerState == null) {
                circuitBreakerState = registerCircuitBreaker(applicationName, invocationTarget, fullMethodSignature, 
                        circuitBreaker);
            }
        }
```
3. The  [FaultToleranceService.registerCircuitBreaker()](https://github.com/payara/Payara/blob/bc93c5e8948d3cb4a94c321a407d2589cf9cf074/appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceService.java#L415) uses the requestVolumeThreshold value from the annotation directly instead of using the read from the configuration file as the following: -

```java
faultToleranceObjects.get(applicationName).getCircuitBreakerStates().get(invocationTarget)
                    .put(fullMethodSignature, new CircuitBreakerState(circuitBreaker.requestVolumeThreshold()));
```

I hope this information may be useful for reproducing the issue.

**Comment by charleech** [comment]
Hi, 

I've create a reproducible test case at

https://github.com/charleech/payara-evaluation/tree/master/payara-evaluation-fault-tolerance

I hope this may be useful.

**Comment by jbee** [comment]
Hi, this will be fixed in connection with the work I do on MP FT 2.0.

**Comment by jbee** [comment]
Will be fixed by #3911

**Comment by smillidge** [comment]
Fixed by #3911
